TESTS/mbed_platform/minimal-printf/compliance/main.cpp

```diff
@@ -42,6 +42,94 @@
 
 using namespace utest::v1;
 
+#if defined(__NEWLIB_NANO)
+
+#define MAX_STRING_SIZE 100
+//Global buffer used by test cases to store string literal
+char expected_string[MAX_STRING_SIZE];
+
+typedef enum {
+    RADIX_10,//radix base for decimal
+    RADIX_16 //radix base for Hexadecimal
```

Comment:
We don't need to specify `RADIX_16 = 16`, it can be any value. But to follow up on Hugues's comment, you should rename `radix_base`  `radix_t` and use `radix_t radix` instead of `radix base`

Reply:
You need it to be what i wrote above so it can work with the code i suggested below.